### PR TITLE
Heal backend configuration file

### LINKED ISSUE #6461: xl: Heal config file
## Expected Behavior
Heal config file should be healed when healing bucket metadata

## Current Behavior
Config file is not healed when healing bucket metadata

## Possible Solution
Just heal config file as it is already done with listener.json and policy.json

## Steps to Reproduce (for bugs)
1. `minio server /tmp/xl/{1..4}`
2. `rm -r /tmp/xl/1/.minio.sys/config/`
3. `mc admin heal myminio/`

## Context
Testing

## Regression
No

## Your Environment
All

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Heal backend configuration file

## Motivation and Context
Fixes #6461

## Regression
No

## How Has This Been Tested?
- minio server /tmp/xl/{1..4}
- rm -r /tmp/xl/1/.minio.sys/config/
- mc admin heal myminio/

The last command should fix the missing config

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/admin-heal-ops.go [hunk 1]
```diff
@@ -531,6 +535,29 @@ func (h *healSequence) traverseAndHeal() {
 	close(h.traverseAndHealDoneCh)
 }
 
+// healConfig - heals config.json, retrun value indicates if a failure occurred.
```
Comment (vadmeste) [inline cmd/admin-heal-ops.go:538]: s/retrun/return

File: cmd/admin-heal-ops.go [hunk 2]
```diff
@@ -531,6 +535,29 @@ func (h *healSequence) traverseAndHeal() {
...
+	}
+
+	configFile := path.Join(minioConfigPrefix, minioConfigFile)
+	configBackupFile := path.Join(minioConfigPrefix, minioConfigBackupFile)
+	for _, cfg := range []string{configFile, configBackupFile} {
+		res, err := objectAPI.HealObject(h.ctx, minioMetaBucket, cfg, h.settings.DryRun)
+		if err != nil {
+			return err
+		}
+		res.Type = madmin.HealItemBucketMetadata
```
Comment (vadmeste) [inline cmd/admin-heal-ops.go:553]: It seems config is not exactly a bucket metadata, let's add deployment metadata if possible.

**Review by vadmeste — DISMISSED** [review]

File: cmd/admin-heal-ops.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/admin-heal-ops.go:553]: Will add it later, for now it is repurposed in this manner. It requires a release of mc etc to get this all working which I didn't see as a good idea.

Comment (vadmeste) [reply cmd/admin-heal-ops.go:553]: > Will add it later, for now it is repurposed in this manner. It requires a release of mc etc to get this all working which I didn't see as a good idea.

okay, one last thing.. it seems that config backup is not there, and I guess we should not return an error if it is not found

Comment (harshavardhana) [reply cmd/admin-heal-ops.go:553]: Does it return an error ?

Comment (vadmeste) [reply cmd/admin-heal-ops.go:553]: > Does it return an error ?

Yes, this is what mc shows currently
```
Heal had an error -  Object not found: .minio.sys#config/config.json.backup
```

Comment (harshavardhana) [reply cmd/admin-heal-ops.go:553]: Looks like that should be ignored by HealObject anycase @vadmeste

**Review by vadmeste — APPROVED** [review]
LGTM & tested

**Review by nitisht — APPROVED** [review]

**Pull request merged**